android/java/org/chromium/chrome/browser/onboarding/SearchEngineOnboardingFragment.java

```diff
@@ -77,6 +77,7 @@ public View onCreateView(
     private void refreshData() {
         TemplateUrlService templateUrlService =
                 BraveTemplateUrlServiceFactory.getForProfile(mProfile);
+        assert templateUrlService != null; // Non-private profile shall always have the service
```

Comment:
we need to handle release version

Reply:
are we 100% sure that private profile won't appear here?

Reply:
Because `mProfile = Profile.getLastUsedRegularProfile()` so it will always be regular profile. I added a null check after.